### PR TITLE
Update CallMedia, CallConnection, and CallRecording methods and models based on ABR feedback

### PR DESCRIPTION
**Pull request by atazimsft**
CallMedia updates:
- Update all <MethodName>Options param names to just be options
- Change CallMediaRecognizeDtmfOptions.StopTones to IList<DtmfTone>
- Rename FileSource to MediaFileSource
- Rename ReasonCode to MediaEventReasonCode
- Rename StartRecognizingResult to StartRecognizingCallMediaResult

CallConnection updates:
- Update GetParticipant() function to accept CommunicationIdentifier
- Change targetCallInvite to target
- Use IList<T> instead of IReadOnlyList<T> for UnmuteParticipantsOptions

CallRecording updates:
- Remove the word "Recording" from the end of all functions in CallRecording client
- Change any occurrence of the word "Endpoint" to "Uri"

### CODE REVIEW

**Comment by ikumarapeli-msft** [comment]
has this been addressed 

All | C# L11 | Key Credential may be missing for SDKs | Update Service Client to accept key credential + Test

File: sdk/communication/Azure.Communication.CallAutomation/src/CallMedia.cs [hunk 1]
```diff
@@ -46,15 +46,15 @@ protected CallMedia()
         /// <param name="playSource"></param>
         /// <param name="cancellationToken"></param>
         /// <param name="playTo"></param>
-        /// <param name="playOptions"></param>
+        /// <param name="options"></param>
         /// <returns>Returns <see cref="PlayResult"/>, which can be used to wait for Play's related events.</returns>
-        public virtual async Task<Response<PlayResult>> PlayAsync(PlaySource playSource, IEnumerable<CommunicationIdentifier> playTo, PlayOptions playOptions = default, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<PlayResult>> PlayAsync(PlaySource playSource, IEnumerable<CommunicationIdentifier> playTo, PlayOptions options = default, CancellationToken cancellationToken = default)
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallMedia.cs:51]: This signature should be either only support simple method, or complex method (options pattern), but not both.
We should have overloads of 
1. Simple method (only have simple inputs, the one that only requires required input)
2. Only options as input

**Comment by minwoolee-msft** [comment]
I remember there was feedback on:
> public virtual CommunicationUserIdentifier GetSourceIdentity

This should be property, instead of method and allow (C# L27)

**Comment by minwoolee-msft** [comment]
We should get rid of all client validaiton message like these:
> public static class CallAutomationErrorMessages

No client validation as discussed (C# L194)

**Comment by minwoolee-msft** [comment]
Update on all event names (C# L160):
AddParticipantFailed -> AddParticipantFailedEventData
CallAutomationEventBase -> CallAutomationEventData

File: sdk/communication/Azure.Communication.CallAutomation/src/CallMedia.cs [hunk 2]
```diff
@@ -41,20 +41,18 @@ protected CallMedia()
...
-        /// <param name="playSource"></param>
-        /// <param name="cancellationToken"></param>
-        /// <param name="playTo"></param>
-        /// <param name="playOptions"></param>
-        /// <returns>Returns <see cref="PlayResult"/>, which can be used to wait for Play's related events.</returns>
-        public virtual async Task<Response<PlayResult>> PlayAsync(PlaySource playSource, IEnumerable<CommunicationIdentifier> playTo, PlayOptions playOptions = default, CancellationToken cancellationToken = default)
+        /// <param name="cancellationToken">An optional CancellationToken to cancel the request.</param>
+        /// <param name="options">An optional object containing play options and configurations.</param>
+        /// <returns>Returns a <see cref="PlayResult"/> object, which can be used to wait for Play's related events.</returns>
+        public virtual async Task<Response<PlayResult>> PlayAsync(PlayOptions options, CancellationToken cancellationToken = default)
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallMedia.cs:49]: is there simple method for Play & PlayToAll?
These 2 methods could be a good candidate for having both simple & complex method - we have complex method with option pattern, but we could also use simple method for better usability.

You can take a look at CreateCall and its overload methods for how it implemented simple & complex methods.

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CallMediaRecognizeDtmfOptions.cs [hunk 3]
```diff
@@ -34,6 +34,6 @@ public CallMediaRecognizeDtmfOptions(CommunicationIdentifier targetParticipant,
         /// <summary>
         /// List of tones that will stop recognizing.
         /// </summary>
-        public IReadOnlyList<DtmfTone> StopTones { get; set; }
+        public IList<DtmfTone> StopTones { get; set; }
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/CallMediaRecognizeDtmfOptions.cs:37]: IEnumerable<DtmfTone>

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/PlayOptions.cs [hunk 4]
```diff
@@ -1,17 +1,42 @@
...
-        /// <summary> The option to play the provided audio source in loop when set to true. </summary>
+        /// <summary>
+        /// A PlaySource object representing the source to play.
+        /// </summary>
+        public PlaySource PlaySource { get; }
+
+        /// <summary>
+        /// A list of target identifiers to play the file to.
+        /// </summary>
+        public IEnumerable<CommunicationIdentifier> PlayTo { get; }
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/PlayOptions.cs:21]: IReadOnlyList

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/PlayToAllOptions.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        /// </summary>
+        public bool Loop { get; set; }
+
+        /// <summary>
+        /// The Operation Context.
+        /// </summary>
+        public string OperationContext { get; set; }
+
+        /// <summary>
+        /// Creates a new PlayOptions object.
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/PlayToAllOptions.cs:29]: PlayToAllOptions

**Review by minwoolee-msft — APPROVED** [review]

**Pull request merged**